portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,16 +63,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
 
     response.setContentType("application/json");
-    String json = GSON.toJson(comments);
-    response.getWriter().println(json);
+    if (userService.isUserLoggedIn()) {
+      response.getWriter().println(GSON.toJson(comments));
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/index.html";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+      String loginMessage = "<p>You must be logged in to view comments.</p>"
+          + "<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>";
+      response.getWriter().println(GSON.toJson(loginMessage));
```

Comment:
from the code, logged out users can't view comments at the moment. would it be better to show logged out users the comments, but disable the ability to post a comment?

Reply:
done.